compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/BIRPackageSymbolEnter.java

```diff
@@ -1453,7 +1485,12 @@ public BType readType(int cpI) throws IOException {
                     }
                     int funcCount = inputStream.readInt();
                     for (int i = 0; i < funcCount; i++) {
-                        ignoreAttachedFunc();
+                        //populate intersection type object functions
+                        if (isImmutable(objectSymbol.flags) && Symbols.isFlagOn(flags, Flags.ANONYMOUS)) {
+                            populateIntersectionTypeReferencedFunctions(inputStream, objectSymbol);
```

Comment:
Shall we create an issue for these methods not being available as top level functions also? We need to verify if they should be added.

Reply:
https://github.com/ballerina-platform/ballerina-lang/issues/30051 tracks this